### ISSUE TITLE
Best power too rich of peak EGT

### ISSUE THREAD

**Issue by algefaen** [c172p-team/c172p#1128]
Hi! I might be entirely wrong, but it seems like our Cessna has a wrong "best power" lean settings compared to the Lycoming manuals. From their charts it looks we should get "max power" at about 130F rich of peak EGT, while my experiments in the sim give me max power at around 250F rich of peak EGT. The sim also seems to be losing power faster than my (anecdotal) real life experience of leaning the C172.

It's very possible that this is a JSBSim issue and not a problem in the c172p code.

Tested using a3e58633 of the master branch.

**Comment by dany93** [comment]
Hi @algefaen,
I totally ignore how these "best power" and EGT are calculated by JSBSim.
Influence of bsfc? [JSBSim Engines](http://wiki.flightgear.org/JSBSim_Engines)

Also, I have zero knowledge of this IRL.

**Comment by wkitty42** [comment]
On 06/19/2018 08:07 PM, algefaen wrote:
> It's very possible that this is a JSBSim issue and not a problem in the c172p
> code.

FWIW: JSBSim has just been updated in FG (on 2018 06 19)...

for a real fun time, call 8675309... errr... oops... update, build, and run FG 
from next and the latest c172p from this repo ;)

**Comment by dany93** [comment]
@algefaen 
At which altitude?
I don't understand how to read the EGT graduations on the instrument. I guess temperature differences? From which reference?

Have you observed the 
`engines/active-engine/egt-degf`?
or (same values)
`fdm/jsbsim/propulsion/engine[1]/egt-degF`?
(engine1] or [0])

**Comment by gilbertohasnofb** [comment]
Someone could also test the c182s and see if the issue is also present there.

**Comment by algefaen** [comment]
I tried at 5k and 10k MSL. Looking at  `engines/engine/egt-degf` and `engines/engine/thrust_lb`. (Or `engine[1]` for the 180HP, I guess this is equivalent to looking at `active-engine`. Procedure: Trim out for full throttle flight. Lean to find and write down peak egt (it's pretty easy). Then enrichen to find the maximum thrust_lb and note down the egt-degf again. For me the egt-degf was around 250F lower, whereas it looks like it should be ~130F lower on the charts.

If this is deep in JSBSim territory then we probably have bigger fish to fry. (=

**Comment by dany93** [comment]
I checked and found values close to yours.

A calculation can be found in JSBSim Flight Dynamics Model, [FGPiston.cpp](http://jsbsim.sourceforge.net/JSBSim/FGPiston_8cpp_source.html)
lines 817 -
`FGPiston::doEGT`
(hard to understand...)

**Comment by algefaen** [comment]
Yeah, that code is a certainly not the easiest to dive into. There's a bit too many magic numbers and tables, so it feels like reading code that's written based on good sources (a book, a good article) but without having the source to make it understandable.

**Comment by algefaen** [comment]
I'll open a JSBSim ticket to get their input...

**Comment by algefaen** [comment]
JSBSim ticket: https://github.com/JSBSim-Team/jsbsim/issues/96

**Comment by gilbertohasnofb** [comment]
@algefaen @dany93 any news on this one? Is there anything we can do on our side or is this indeed a problem with JSBSim?

**Comment by gilbertohasnofb** [comment]
@algefaen any luck with the team of JSBSim? Do you know if they fixed the issue? The ticket seems to still be opened.

**Comment by tonghuix** [comment]
Same issue, when lean mixture on ground: set RPM 1200, and I try to lean for max RPM, but when I pull the mixture lever out slowly, the RPM drops all the way down, no any increasing at all!

**Comment by dany93** [comment]
@tonghuix 
Which altitude?

**Comment by tonghuix** [comment]
> @tonghuix
> Which altitude?

Sorry I forgot mention it, I test it on ground of PHTO, field elevation is 37ft MSL.

**Comment by dany93** [comment]
So there is no need to lean the mixture?

Having to lean for max RPM is only at high altitude, isn't it? (I have no practice of it)

**Comment by tonghuix** [comment]
> So there is no need to lean the mixture?
> 
> Having to lean for max RPM is only at high altitude, isn't it? (I have no practice of it)

Yep, to reduce spark plug fouling chance. Here is a article: https://www.avweb.com/ownership/leaning-on-the-ground/

**Comment by hbeni** [comment]
(related to #1492 )
